### PR TITLE
VSF2-60 - Add Withlist support to GraphQL

### PR DESCRIPTION
**Pull request by Wojdylak**

### CODE REVIEW

File: src/CommandHandler/Wishlist/AddItemToWishlistHandler.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+use Symfony\Component\Messenger\Handler\MessageHandlerInterface;
+use Webmozart\Assert\Assert;
+
+final class AddItemToWishlistHandler implements MessageHandlerInterface
+{
+    public const EVENT_NAME = 'bitbag.sylius_vue_storefront2.add_item_to_wishlist.complete';
+
+    private WishlistRepositoryInterface $wishlistRepository;
+    private WishlistProductFactoryInterface $wishlistProductFactory;
+    private IriConverterInterface $iriConverter;
```
Comment (senghe) [inline src/CommandHandler/Wishlist/AddItemToWishlistHandler.php:30]:
```suggestion
    private WishlistRepositoryInterface $wishlistRepository;
    
    private WishlistProductFactoryInterface $wishlistProductFactory;
    
    private IriConverterInterface $iriConverter;
```

File: src/CommandHandler/Wishlist/CreateNewWishlistHandler.php [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+ * This file was created by developers working at BitBag
+ * Do you need more information about us and what we do? Visit our https://bitbag.io website!
+ * We are hiring developers from all over the world. Join us and start your new, exciting adventure and become part of us: https://bitbag.io/career
+*/
+
+declare(strict_types=1);
+
+namespace BitBag\SyliusVueStorefront2Plugin\CommandHandler\Wishlist;
+
+
```
Comment (senghe) [inline src/CommandHandler/Wishlist/CreateNewWishlistHandler.php:13]:
```suggestion

```

File: src/CommandHandler/Wishlist/RemoveItemFromWishlistHandler.php [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+use Symfony\Component\EventDispatcher\GenericEvent;
+use Symfony\Component\Messenger\Handler\MessageHandlerInterface;
+use Webmozart\Assert\Assert;
+
+final class RemoveItemFromWishlistHandler implements MessageHandlerInterface
+{
+    public const EVENT_NAME = 'bitbag.sylius_vue_storefront2.remove_item_from_wishlist.complete';
+
+    private ObjectManager $wishlistManager;
+    private IriConverterInterface $iriConverter;
```
Comment (senghe) [inline src/CommandHandler/Wishlist/RemoveItemFromWishlistHandler.php:28]:
```suggestion
    private ObjectManager $wishlistManager;
    
    private IriConverterInterface $iriConverter;
```

File: src/CommandHandler/Wishlist/RemoveItemFromWishlistHandler.php [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    {
+        /** @var WishlistInterface|null $wishlist */
+        $wishlist = $this->iriConverter->getItemFromIri($command->getId());
+        Assert::notNull($wishlist);
+
+        /** @var ProductVariantInterface|null $productVariant */
+        $productVariant = $this->iriConverter->getItemFromIri($command->getProductVariant());
+        Assert::notNull($productVariant);
+
+
```
Comment (senghe) [inline src/CommandHandler/Wishlist/RemoveItemFromWishlistHandler.php:52]:
```suggestion

```

File: src/Doctrine/Orm/Extension/WishlistsCurrentUserExtension.php [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+use Doctrine\ORM\QueryBuilder;
+use Sylius\Component\Core\Model\ShopUserInterface;
+use Symfony\Component\Security\Core\Security;
+
+final class WishlistsCurrentUserExtension implements QueryCollectionExtensionInterface
+{
+    const GRAPHQL_OPERATION_KEY = 'graphql_operation_name';
+    const COLLECTION_OPERATION_KEY = 'collection_operation_name';
+    const WISHLIST_GRAPHQL_OPERATION_NAME = 'collection_query';
+    const WISHLIST_COLLECTION_OPERATION_NAME = 'shop_get_wishlists';
```
Comment (senghe) [inline src/Doctrine/Orm/Extension/WishlistsCurrentUserExtension.php:27]:
```suggestion
    const GRAPHQL_OPERATION_KEY = 'graphql_operation_name';
    
    const COLLECTION_OPERATION_KEY = 'collection_operation_name';
    
    const WISHLIST_GRAPHQL_OPERATION_NAME = 'collection_query';
    
    const WISHLIST_COLLECTION_OPERATION_NAME = 'shop_get_wishlists';
```

File: src/Resources/api_resources/ChannelPricing.xml [hunk 6]
```diff
@@ -11,9 +11,16 @@ We are hiring developers from all over the world. Join us and start your new, ex
...
     <resource class="%sylius.model.channel_pricing.class%" shortName="ChannelPricing">
+        <attribute name="filters">
+            <attribute>bit_bag.sylius_vue_storefront2plugin.filter.channel_pricing_channel_code_filter</attribute>
+        </attribute>
+
         <graphql>
             <operation name="collection_query">
                 <attribute name="pagination_type">page</attribute>
+                <attribute name="filters">
+                    <attribute>bit_bag.sylius_vue_storefront2plugin.filter.channel_pricing_channel_code_filter</attribute>
```
Comment (senghe) [inline src/Resources/api_resources/ChannelPricing.xml:22]:
```suggestion
                    <attribute>bitbag.sylius_vue_storefront2plugin.filter.channel_pricing_channel_code_filter</attribute>
```

File: src/Resources/api_resources/ChannelPricing.xml [hunk 7]
```diff
@@ -11,9 +11,16 @@ We are hiring developers from all over the world. Join us and start your new, ex
            xsi:schemaLocation="https://api-platform.com/schema/metadata https://api-platform.com/schema/metadata/metadata-2.0.xsd"
 >
     <resource class="%sylius.model.channel_pricing.class%" shortName="ChannelPricing">
+        <attribute name="filters">
+            <attribute>bit_bag.sylius_vue_storefront2plugin.filter.channel_pricing_channel_code_filter</attribute>
```
Comment (senghe) [inline src/Resources/api_resources/ChannelPricing.xml:15]:
```suggestion
            <attribute>bitbag.sylius_vue_storefront2plugin.filter.channel_pricing_channel_code_filter</attribute>
```

File: src/Resources/services/command_handlers.xml [hunk 8]
```diff
@@ -141,5 +141,41 @@ We are hiring developers from all over the world. Join us and start your new, ex
             <tag name="messenger.message_handler" bus="sylius_default.bus" />
         </service>
 
+        <service
+            id="bit_bag.sylius_vue_storefront2plugin.command_handler.wishlist.create_new_wishlist_handler"
```
Comment (senghe) [inline src/Resources/services/command_handlers.xml:145]: Could you keep service ids consistent?

File: src/Validator/UniqueNameShopUserWishlistValidator.php [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+
+final class UniqueNameShopUserWishlistValidator extends ConstraintValidator
+{
+    private Security $security;
+    private WishlistRepositoryInterface $wishlistRepository;
+    private WishlistNameCheckerInterface $wishlistNameChecker;
```
Comment (senghe) [inline src/Validator/UniqueNameShopUserWishlistValidator.php:26]:
```suggestion
    private Security $security;
    
    private WishlistRepositoryInterface $wishlistRepository;
    
    private WishlistNameCheckerInterface $wishlistNameChecker;
```

File: tests/Application/config/packages/validator.yaml [hunk 10]
```diff
@@ -1,3 +1,6 @@
 framework:
     validation:
         enable_annotations: true
+        mapping:
+            paths:
+                - '%kernel.project_dir%/../../src/Resources/validation/'
```
Comment (senghe) [inline tests/Application/config/packages/validator.yaml:6]: This information has to be put into installation docs

File: tests/Application/config/bundles.php [hunk 11]
```diff
@@ -60,4 +60,5 @@
     BitBag\SyliusVueStorefront2Plugin\BitBagSyliusVueStorefront2Plugin::class => ['all' => true],
     Gesdinet\JWTRefreshTokenBundle\GesdinetJWTRefreshTokenBundle::class => ['all' => true],
     Sylius\Calendar\SyliusCalendarBundle::class => ['all' => true],
+    BitBag\SyliusWishlistPlugin\BitBagSyliusWishlistPlugin::class => ['all' => true],
```
Comment (senghe) [inline tests/Application/config/bundles.php:63]: This information has to be put into installation docs

**Review by senghe — CHANGES_REQUESTED** [review]

File: src/Resources/services/command_handlers.xml (hunk 8 above)
Comment (Wojdylak) [reply src/Resources/services/command_handlers.xml:145]: Yes :)

**Review by senghe — APPROVED** [review]

**Pull request merged**